src/capi/azext_capi/templates/KubeadmConfigTemplate.yaml

```diff
@@ -14,11 +14,15 @@ spec:
         sshAuthorizedKeys:
         - {{AZURE_SSH_PUBLIC_KEY}}
         {%- endif %}
-      preKubeadmCommands: []
-      postKubeadmCommands:
-        - nssm set kubelet start SERVICE_AUTO_START
-        - powershell C:/defender-exclude-calico.ps1
       {%- endif %}
+      preKubeadmCommands:{% if not PRE_BOOTSTRAP_CMDS %} []{% endif %}
```

Comment:
@mboersma, I think for this scenario might be beneficial to omit these. As if we do leave these empty in the template. Wouldn't we want to do the same for the rest of the fields in template? And also doing the connection between pre/postKubeadmCommands to bootstrap commands, it might not be as straightforward for users not familiar with Kubeadm.

Reply:
I think we should omit the whole section(s) if no bootstrap commands were provided. Having an empty value isn't consistent with the way we handle most other similar cases, and it would make the template a bit easier to read.